var/spack/repos/builtin/packages/conduit/package.py

```diff
@@ -408,18 +408,23 @@ def create_host_config(self, spec, prefix, py_site_pkgs_dir=None):
                     cfg.write(cmake_cache_entry("CMAKE_Fortran_COMPILER_ID",
                                                 "XL"))
 
-                if 'xl@coral' in os.getenv('SPACK_COMPILER_SPEC', ""):
+                if (f_compiler is not None) and ("xlf" in f_compiler):
                     # Fix missing std linker flag in xlc compiler
                     flags = "-WF,-C! -qxlf2003=polymorphic"
                     cfg.write(cmake_cache_entry("BLT_FORTRAN_FLAGS",
                                                 flags))
                     # Grab lib directory for the current fortran compiler
                     libdir = os.path.join(os.path.dirname(
                                           os.path.dirname(f_compiler)), "lib")
-                    flags  = "${BLT_EXE_LINKER_FLAGS} -lstdc++ "
-                    flags += "-Wl,-rpath,{0} -Wl,-rpath,{0}64".format(libdir)
+                    rpaths = "-Wl,-rpath,{0} -Wl,-rpath,{0}64".format(libdir)
+
+                    flags  = "${BLT_EXE_LINKER_FLAGS} -lstdc++ " + rpaths
                     cfg.write(cmake_cache_entry("BLT_EXE_LINKER_FLAGS",
                                                 flags))
+                    if "+shared" in spec:
+                        flags = "${CMAKE_SHARED_LINKER_FLAGS} " + rpaths
+                        cfg.write(cmake_cache_entry(
+                                  "CMAKE_SHARED_LINKER_FLAGS", flags))
```

Comment:
Wow, compilers are being very mean to you -- this looks good, thanks for working this out.